core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -1306,23 +1326,46 @@ class GroupCoordinator(val brokerId: Int,
                 protocolType = group.protocolType,
                 protocolName = group.protocolName,
                 leaderId = currentLeader,
+                skipAssignment = false,
                 error = error
               ))
+            } else if (supportSkippingAssignment) {
+              // Starting from version 9 of the JoinGroup API, static members are able to
+              // skip running the assignor based on the `SkipAssignment` field. We leverage
+              // this to tell the leader that it is the leader of the group but by skipping
+              // running the assignor while the group is in stable state.
```

Comment:
Can you mention somewhere in the comments that this approach still doesn't fully handle a condition if metadata has changed while the leader was down?